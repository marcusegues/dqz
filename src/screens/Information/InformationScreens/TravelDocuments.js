// @flow
import React from 'react';
import { Text, View } from 'react-native';
import type { ComponentType } from 'react';
import { translate } from 'react-i18next';

class TravelDocumentsInner extends React.Component<{
  t: TFunction,
  navigation: Navigation,
}> {
  // static navigationOptions = ({ screenProps }) => ({
  //   headerTitle: (
  //     <HeaderTitle text={screenProps.t('information:informationTitle')} />
  //   ),
  // });

  render() {
    const { t, navigation, infoCategory } = this.props;
    return (
      <View>
        <Text>Travel Documents</Text>
      </View>
    );
  }
}

export const TravelDocuments = (translate(['information'])(
  TravelDocumentsInner
): ComponentType<{}>);
