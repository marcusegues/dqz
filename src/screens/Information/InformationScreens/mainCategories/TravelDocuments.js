// @flow
import React from 'react';
import type { ComponentType } from 'react';
import { translate } from 'react-i18next';
// $FlowFixMe
import type { TFunction } from '../../../../types/generalTypes';
import { AppInfoSubText } from '../../../AppInfo/subComponents/AppInfoSubText';
import { SubCategoryContainer } from '../../subComponents/SubCategoryContainer';

type TravelDocumentsInnerProps = {
  t: TFunction,
};

const TravelDocumentsInner = ({ t }: TravelDocumentsInnerProps) => (
  <SubCategoryContainer title={t('travelDocumentsMainText')}>
    <AppInfoSubText text={t('travelDocumentsText1')} />
    <AppInfoSubText
      text="www.sem.admin.ch"
      style={{
        textDecorationLine: 'underline',
        textDecorationStyle: 'solid',
      }}
    />
    <AppInfoSubText text={t('travelDocumentsText2')} />
    <AppInfoSubText
      text="www.wcoomd.org"
      style={{
        textDecorationLine: 'underline',
        textDecorationStyle: 'solid',
      }}
    />
  </SubCategoryContainer>
);

export const TravelDocuments = (translate(['travelDocuments'])(
  TravelDocumentsInner
): ComponentType<TravelDocumentsInnerProps>);
