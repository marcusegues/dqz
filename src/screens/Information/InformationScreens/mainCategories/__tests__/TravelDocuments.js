import React from 'react';
import { shallow } from 'enzyme';
import { TravelDocuments } from '../TravelDocuments';

jest.mock('react-i18next', () => ({
  translate: () => Component => props => <Component t={() => ''} {...props} />,
}));

const clickFn = jest.fn();

describe('TravelDocuments', () => {
  test('TravelDocuments renders as expected', () => {
    const wrapper = shallow(<TravelDocuments />);
    expect(wrapper).toMatchSnapshot();
  });
  test('TravelDocuments renders as expected with language', () => {
    const wrapper = shallow(
      <TravelDocuments
        navigation={{ dispatch: clickFn }}
        i18n={{ language: 'de' }}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  test('TravelDocuments renders as expected with language with dive()', () => {
    const wrapper = shallow(
      <TravelDocuments
        navigation={{ dispatch: clickFn }}
        i18n={{ language: 'de', t: 'test' }}
      />
    );
    expect(wrapper.dive()).toMatchSnapshot();
  });
  // test('TravelDocuments renders as expected with language and with dive()', () => {
  //   //   const wrapper = shallow(
  //   //     <TravelDocuments
  //   //       navigation={{ dispatch: clickFn }}
  //   //       i18n={{ language: 'de', t: 'test' }}
  //   //     />
  //   //   );
  //   //   const render = wrapper.dive();
  //   //   render
  //   //     .find(<AppInfoLink text="test" onPress={clickFn} />)
  //   //     .forEach(child => {
  //   //       child.simulate('press');
  //   //     });
  //   //   expect(clickFn.mock.calls.length).toBe(1);
  //   // });
});
