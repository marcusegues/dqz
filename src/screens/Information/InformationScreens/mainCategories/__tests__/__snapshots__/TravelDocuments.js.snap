// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TravelDocuments TravelDocuments renders as expected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TravelDocumentsInner
    t={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <AppInfoSubText
          greyBox={false}
          style={Object {}}
          text=""
        />,
        <AppInfoLink
          onPress={[Function]}
          style={Object {}}
          text="www.sem.admin.ch"
        />,
        <AppInfoSubText
          greyBox={false}
          style={Object {}}
          text=""
        />,
        <AppInfoLink
          onPress={[Function]}
          style={Object {}}
          text="www.wcoomd.org"
        />,
      ],
      "title": "",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "greyBox": false,
          "style": Object {},
          "text": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "onPress": [Function],
          "style": Object {},
          "text": "www.sem.admin.ch",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "greyBox": false,
          "style": Object {},
          "text": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "onPress": [Function],
          "style": Object {},
          "text": "www.wcoomd.org",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <AppInfoSubText
            greyBox={false}
            style={Object {}}
            text=""
          />,
          <AppInfoLink
            onPress={[Function]}
            style={Object {}}
            text="www.sem.admin.ch"
          />,
          <AppInfoSubText
            greyBox={false}
            style={Object {}}
            text=""
          />,
          <AppInfoLink
            onPress={[Function]}
            style={Object {}}
            text="www.wcoomd.org"
          />,
        ],
        "title": "",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "greyBox": false,
            "style": Object {},
            "text": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "onPress": [Function],
            "style": Object {},
            "text": "www.sem.admin.ch",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "greyBox": false,
            "style": Object {},
            "text": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "onPress": [Function],
            "style": Object {},
            "text": "www.wcoomd.org",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
